collect_app/src/test/java/org/odk/collect/android/forms/FormsRepositoryTest.java

```diff
@@ -20,26 +20,43 @@
     public abstract String getFormFilesPath();
 
     @Test
-    public void getOneByFormIdAndVersion_whenFormHasNullVersion_returnsForm() {
+    public void getLatestByFormIdAndVersion_whenFormHasNullVersion_returnsForm() {
         FormsRepository formsRepository = buildSubject();
-        formsRepository.save(buildForm(1L, "1", null, getFormFilesPath())
+        formsRepository.save(buildForm("1", null, getFormFilesPath())
                 .build());
 
-        Form form = formsRepository.getOneByFormIdAndVersion("1", null);
+        Form form = formsRepository.getLatestByFormIdAndVersion("1", null);
         assertThat(form, notNullValue());
         assertThat(form.getId(), is(1L));
     }
 
+    @Test
+    public void getLatestByFormIdAndVersion_whenMultipleExist_returnsLatest() throws InterruptedException {
+        FormsRepository formsRepository = buildSubject();
+        formsRepository.save(buildForm("1", "1", getFormFilesPath())
+                .build());
+        Thread.sleep(2); // ensure that the in-memory inserts don't all get the same timestamp
```

Comment:
Thanks, that is better. See what you think of my implementation. We can get away with overriding the app dependency module just for the test we need it because injection is delayed for providers. It's a little quirky but feels ok to me. I wanted to make sure the same mock was shared between the two repository implementations.